lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -1605,6 +1613,79 @@ bool QuakeBridgeVisitor::VisitCallExpr(clang::CallExpr *x) {
                                          reportNegateError, isAdjoint,
                                          isControl, /*paramCount=*/3);
 
+    // See if this is a custom unitary.
+    std::string maybeUnitaryGenerator = funcName.str() + "_generator_";
+    // Extract number of targets
+    size_t targetCount = 0;
+    std::string genFuncName;
+    for (auto name : customOperationNames) {
+      if (name.first.find(maybeUnitaryGenerator) != std::string::npos) {
+        targetCount =
+            std::stoi(name.first.substr(maybeUnitaryGenerator.length()));
+        genFuncName = getCudaqKernelName(name.second);
+        break;
+      }
+    }
+    if (targetCount > 0) {
+      // Extract number of parameters
+      std::size_t paramCount = [&]() {
+        std::size_t count = 0;
+        for (auto arg : args) {
+          auto argTy = arg.getType();
+          if (isa<FloatType>(argTy))
+            count++;
+          else if (auto ptrTy = dyn_cast<cc::PointerType>(argTy)) {
+            if (isa<FloatType>(ptrTy.getElementType()))
+              count++;
+          }
+        }
+        return count;
+      }();
+      auto srefAttr = SymbolRefAttr::get(
+          StringAttr::get(builder.getContext(), genFuncName));
+      ValueRange operands(args);
+      assert(operands.size() >= 1 && "must be at least 1 operand");
+      if ((operands.size() == 1) &&
+          operands[0].getType().isa<quake::VeqType>()) {
+        auto target = operands[0];
+        if (!negations.empty())
```

Comment:
Rather than block copying buildOp, buildOp should be generalized to allow for the symbol to be an optional argument.

Reply:
The `create` function for `CustomUnitarySymbolOp` differs which resulted in multiple conditions in the `buildOp` function, hence kept it separate here.

Reply:
It differs by one argument, right?

That difference is small as compared to block copying all this functionality into a method that is already too large.